include/ebpf_api.h

```diff
@@ -348,6 +354,114 @@ extern "C"
     void
     ebpf_api_map_info_free(uint16_t map_count, _In_count_(map_count) const ebpf_map_information_t* map_info);
 
+    /**
+     * @brief Load eBPF programs from an ELF file based on default load
+     * attributes. This API does the following:
+     * 1. Read the ELF file.
+     * 2. Create maps.
+     * 3. Load all programs.
+     * 4. Return fd to the first program.
+     *
+     * If the caller supplies a program type and/or attach type, that
+     * supplied value takes precedence over the derived program/attach type.
+     *
+     * @param[in] file_name ELF file name with full path.
+     * @param[in] program_type Optionally, the program type to use when loading
+     *  the eBPF program. If program type is not supplied, it is derived from
+     *  the section prefix in the ELF file.
+     * @param[in] attach_type Optionally, the attach type to use for the loaded
+     *  eBPF program. If attach type is not supplied, it is derived from the
+     *  section prefix in the ELF file.
+     * @param[out] ebpf_object Returns pointer to ebpf_object object. The caller
+        is expected to call ebpf_object_close() at the end.
+     * @param[out] program_fd Returns a file descriptor for the first program.
+     *  The caller should not call _close() on the fd, but should instead use
+     *  ebpf_object_close() to close this (and other) file descriptors.
+     * @param[out] log_buffer Returns a pointer to a null-terminated log buffer.
+     *  The caller is responsible for freeing the returned log_buffer pointer
+     *  by calling ebpf_api_free_string().
+     *
+     * @retval EBPF_SUCCESS The programs are loaded and maps are created successfully.
+     * @retval EBPF_INVALID_ARGUMENT One or more parameters are incorrect.
+     * @retval EBPF_NO_MEMORY Out of memory.
+     * @retval EBPF_ELF_PARSING_FAILED Failure in parsing ELF file.
+     * @retval EBPF_FAILED Some other error occured.
+     */
+    _Success_(result == EBPF_SUCCESS) ebpf_result_t ebpf_program_load(
```

Comment:
This is already annotated in ebpf_result.h as having EBPF_SUCCESS meaning success, why duplicate it here?

Reply:
Fixed.